symfit/core/objectives.py

```diff
@@ -124,6 +124,30 @@ def _invariant_kwargs(self):
         )
         return kwargs
 
+    def __eq__(self, other):
+        """
+        Objectives are considered equal if they are of the same type, have the
+        same model, and the same data.
+        """
+        # Class equality is enforced, even though this breaks subclassing.
+        # This is to prevent false positives, which could be way worse. In the
+        # case of subclassing, we leave it up to the subclasser to decide when
+        # equality is achieved.
+        if self.__class__ == other.__class__ and self.model == other.model:
+            # Check if the data is also equivalent
+            for key, value in self.data.items():
+                try:
+                    equal = np.allclose(other.data[key], value)
+                except TypeError:
+                    equal = other.data[key] == value
+                finally:
+                    if not equal:
+                        return False
+            else:
+                return True
+        else:
```

Comment:
Useless `else` statements. Also, invert the condition above: `if not equal classes: return False`

Reply:
I'm not sure I follow on the inversion. However, I will remove the last else and do a break of the loop instead, so the for/else really serves a pourpose and `False` is our catch-all.